packages/lesswrong/server/editor/mentions-notify.ts

```diff
@@ -0,0 +1,48 @@
+import * as _ from 'underscore'
+import {createNotifications} from '../notificationCallbacksHelpers'
+import {notificationDocumentTypes} from '../../lib/notificationTypes'
+
+export interface PingbackDocumentPartial {
+  _id: string,
+  pingbacks?: {
+    Users: string[]
+  }
+}
+
+export const notifyUsersAboutMentions = async (currentUser: DbUser, collectionType: string, document: PingbackDocumentPartial, oldDocument?: PingbackDocumentPartial) => {
+  const pingbacksToSend = getPingbacksToSend(currentUser, document, oldDocument)
+
+  // Todo(PR): this works, but not sure if it's generally a correct conversion. 
+  //  TagRels for example won't work, though they don't have content either.
+  //  should we define an explicit mapping?
+  const notificationType = collectionType.toLowerCase()
+  if (!canNotify(currentUser, pingbacksToSend) || !notificationDocumentTypes.has(notificationType)) return
```

Comment:
Yeah, that's definitely not an ideal UX! I noted that down in the PR description as a follow-up work

Reply:
It occurs to me, maybe it's just better to leave out this restriction until we've actually built good UI to warn about it. (I don't expect it to get abused immediately, and we can avoid publicizing the feature until the restrictions are built)

(alternately, if there's anything we can do to throw an error here without too much additional dev work, that'd probably be best for now)

Reply:
Plausibly literally throwing exception should be ok. (from what I saw elsewhere that is handled relatively gracefully)

Reply:
![image](https://user-images.githubusercontent.com/81347/214202910-e54158ca-a87c-46e6-95c5-1efc4b60014a.png)
yeah, seems ok

Reply:
Though actually to make that style of error work properly one would need to move the callbacks into the "before" lifecycle instead of after - otherwise weird things happen. 
it seems fine to move it to before though

Reply:
Hrm actually moving to "before" has issues 🤔

Reply:
So the behavior if I were to keep the callback in "after" & throw exception is something like:
- operation gets performed (comment created, post created or updated)
- User will see the exception about pingbacks and not get redirected (so the perception is that operation didn't work) 
- presumably none of the further "after" callbacks got run. 

On balance, even though it's giving user clearer feedback, it adds it's own confusion and bug potential and I wouldn't go there

Reply:
as an 80/20 thing set the limit to be 25 to start with - and we can lower it when better ui support is available